### DISCUSSION TITLE
What does goreleaser do and not do? (Gitlab Releases etc)

### DISCUSSION THREAD

**Discussion by tommyalatalo** [goreleaser/goreleaser#1834] [Q&A] [answered]
I'm currently using semantic-release for release handling of a project of mine, but I'm considering switching over to goreleaser since my project is in go, and goreleaser may seem more suitable.

However I'm confused as to how goreleaser creates Gitlab releases since it's described in the docs that goreleaser doesn't tag the repository? Do you need to use a third party tool such as svu to first tag the commit and then run goreleaser to use the tag and create the Gitlab release?

What is the reasoning behind goreleaser not tagging the repository and creating a release commit?
Are updated changelogs only dropped as artifacts into the Gitlab/Github releases and not committed back into the repository?

**Comment by caarlos0** [comment, marked as answer]
GoReleaser only build and publishes things, it doesn't create tags nor handle changelog files inside the repo.

You can run both semantic-release and goreleaser without a issue in theory, goreleaser will leave the changelog with what semantic-release puts there (unless you delete it) and that's pretty much it.

The reasoning is just reducing scope. GoReleaser does a lot of things, and the tagging thing is already resolved by other tools - and a lot of people just do it manually.
